core/shared/src/main/scala/fs2/Stream.scala

```diff
@@ -392,6 +392,54 @@ final class Stream[+F[_], +O] private (private val free: FreeC[Algebra[Nothing,
       }
     }
 
+  /**
+    * Outputs chunks of size larger than N
+    *
+    * Chunks from the source stream are split as necessary.
+    *
+    * If `allowFewerTotal` is true,
+    * if the stream is smaller than N, should the elements be included
+    *
+    * @example {{{
+    * scala> (Stream(1,2) ++ Stream(3,4) ++ Stream(5,6,7)).chunkMin(3).toList
+    * res0: List[Chunk[Int]] = List(Chunk(1, 2, 3, 4), Chunk(5, 6, 7))
+    * }}}
+    */
+  def chunkMin(n: Int, allowFewerTotal: Boolean = true): Stream[F, Chunk[O]] = {
+    def go[A](accFull: Option[Chunk[A]],
```

Comment:
Instead of `Option[Chunk[A]]`, can you use `Chunk.Queue[A]`? This lets us avoid concatenating each chunk with the accumulator as it is received -- instead, they are all stored in the queue and then can be concatenated/taken from/dropped from as necessary in an efficient way.